src/uct/ib/base/ib_md.c

```diff
@@ -559,21 +559,21 @@ static ucs_status_t uct_ib_memh_dereg(uct_ib_md_t *md, uct_ib_mem_t *memh)
 {
     ucs_status_t s, status = UCS_OK;
 
-    if (memh->flags & UCT_IB_MEM_FLAG_ATOMIC_MR) {
-        memh->flags &= ~UCT_IB_MEM_FLAG_ATOMIC_MR;
+    if (memh->atomic_rkey != UCT_IB_INVALID_MKEY) {
         s = md->ops->dereg_atomic_key(md, memh);
         if (s != UCS_OK) {
             status = s;
         }
     }
+    memh->flags &= ~UCT_IB_MEM_FLAG_ATOMIC_MR;
 
     if (memh->flags & UCT_IB_MEM_FLAG_RELAXED_ORDERING) {
         s = uct_ib_memh_dereg_key(md, memh, UCT_IB_MR_STRICT_ORDER);
-        memh->flags &= ~UCT_IB_MEM_FLAG_RELAXED_ORDERING;
         if (s != UCS_OK) {
             status = s;
         }
     }
+    memh->flags &= ~UCT_IB_MEM_FLAG_RELAXED_ORDERING;
```

Comment:
why need to move?

Reply:
to make it consistent with the previous code block